File: PersonnelManagement/PersonnelManagement/Models/Pmresult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PersonnelManagement.Models
{
    public class Pmresult
    {
        public double Count { get; set; }
        public double Countvar { get; set; }
        public double Ranksexpanded { get; set; }
        public string Ranks { get; set; }  // names of ranks separated by @
        public string Ranksvar { get; set; } // names of ranks separated by @
        public string Maxcount { get; set; } // 160 separated by @
        public string Absolutecount { get; set; } // 180 separated by @ 
        public string Defaultcount { get; set; } // 100, separated by @
        public string Defaultcountvar { get; set; } // 100, separated by @
        public string Mincount { get; set; } // 80, separated by @
        public string Uprank { get; set; } // При наличии ученой степени=1:5,при наличии классности=1:10;2:20@
        public string Uprankready { get; set; } // Из них при наличии ученой степени степени получить звание майор внутренней службы 10 ед., подполковник внутренней службы 20 ед. & ...
        public string Civil { get; set; } // Государственные служащие=1@Cлужащие=5
        public string Uppedmap { get; set; } // Карта: Название=Айди;На сколько ранков максимум доступен подъем.  При наличии классности=1;2
        public string Uppedcount { get; set; } //При наличии классности=1:100;2:120&При наличии ученой степени=1:5@ 
        public string Uprankunited { get; set; } // Капитан внутренней службы:1:5;Майор внутренней службы:2:5;
        public string Comefromunited { get; set; } // Старший лейтенант внутренней службы:1:5;Капитан внутренней службы:2:5;
        public string Sumunited { get; set; } // 100:120:100,100:100:100@
        public string Unitedlengthmax { get; set; } // Наибольшая из длин uprankunited или comefromunited. 0@3@4@0@1

        public byte[] File { get; set; }

        //public Pmresult()
        //{
        //    Ranks = "";

        //}

    }
}

File: PersonnelManagement/PersonnelManagement/Front/classes/pmresult.ts
export default class Pmresult {
    count: number;
    countvar: number;
    ranksexpanded: number;
    ranks: string;
    ranksvar: string;
    maxcount: string; // 160 separated by , 
    absolutecount: string; // 180 separated by , 
    defaultcount: string; // 100, separated by ,
    defaultcountvar: string; // 100, separated by ,
    mincount: string; // 80, separated by ,
    uprank: string; // При наличии ученой степени=1:5,при наличии классности=1:10;2:20,
    uprankready: string;  // Из них при наличии ученой степени степени получить звание майор внутренней службы 10 ед., подполковник внутренней службы 20 ед. & ...
    civil: string; // Государственные служащие=1,Cлужащие=5
    uppedmap: string; // Карта: Название=Айди;На сколько ранков максимум доступен подъем.  При наличии классности=1;2
    uppedcount: string; //При наличии классности=1:100;2:120&При наличии ученой степени=1:5, 
    uprankunited: string; // Капитан внутренней службы:1:5;Майор внутренней службы:2:5;
    comefromunited: string; // Старший лейтенант внутренней службы:1:5;Капитан внутренней службы:2:5;
    sumunited: string; // 100:120:100,30:50:50,
    unitedlengthmax: string; // Наибольшая из длин uprankunited или comefromunited. 2,3,0,0
    file: Blob;
}